controllers/bootstrap/init.go

```diff
@@ -953,7 +955,33 @@ func CheckClusterType(mgr manager.Manager, ns string) (bool, error) {
 	}
 }
 
-func (b *Bootstrap) DeployCertManagerCR() error {
+// 1. try to get cs-ca-certificate-secret
+// 2. try to get cs-ca-certificate
+// if we get secret but not get the cert, it is BYOC
+func (b *Bootstrap) IsBYOCert() (bool, error) {
+	klog.V(2).Info("Detect if it is BYO cert")
+	secretName := "cs-ca-ceritifcate-secret"
+	secret := &corev1.Secret{}
+	err := b.Client.Get(context.TODO(), types.NamespacedName{Name: secretName, Namespace: b.CSData.ServicesNs}, secret)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return false, err
+		}
+	}
+	certName := "cs-ca-certificate"
+	cert := &certmanagerv1.Certificate{}
+	certerr := b.Client.Get(context.TODO(), types.NamespacedName{Name: certName, Namespace: b.CSData.ServicesNs}, cert)
+	if certerr != nil {
+		if errors.IsNotFound(certerr) {
+			return true, nil
+		}
```

Comment:
This is assuming that secret was found. But what if getting the secret returned `IsNotFound` error? This error is being ignored right now.

Reply:
Yes, if we didn't get this secret, we should return false directly.